include/pybind11/pytypes.h

```diff
@@ -1336,9 +1337,14 @@ class memoryview : public object {
         // Py_buffer uses signed sizes, strides and shape!..
         static std::vector<Py_ssize_t> py_strides { };
```

Comment:
@aldanor 
Question for the original author, who introduced this line (in a different file, later moved here):
https://github.com/pybind/pybind11/commit/a7e62e1ca621c7a7c1b8956873a16c255f2939b6

Why are these std::vector objects static?
Just as an optimization? To minimize the allocation overhead?
Did you consider how this plays with threads? (I don't know if it's possible that multiple threads could get here at the same time, but it's a worry if you have static arrays like here.)

Reply:
I looked briefly at this and don't understand it either. AFAIK the danger is that PyMemoryView_FromBuffer expects the `Py_buffer` to remain valid after the call. (Actually, the precise semantics of this function seem to be a source of confusion in the wider Python community: https://bugs.python.org/issue15821). 

But making the `std::vector` static here won't address this issue. It merely ensures that the pointers survive the return of the function, but then they'll be garbled by the next function call. I think that this whole function needs a fundamental redesign.